### PR TITLE
Implementation of Materialize Grid

### PR DESCRIPTION
**Pull request by 19aanand**
All of the plots are now organized in a Materialize Grid, with the partition selector located in a column on the left-hand side and the plots located in a wider column on the right-hand side. The plots are rebuilt automatically when the user checks/unchecks an option.

### CODE REVIEW

File: main/js/afterSubmit.js [hunk 1]
```diff
@@ -187,16 +187,28 @@ const buildHeatmap = function(expData, clinData) {
...
+  gridRow.classList.add("row");
 
-  // Create div for violin plots
-  addDivInside("violinPlots", "violinLoaderDiv");
+  // Create partition selector div and add it inside Materialize Grid
+  const partitionDivId = "violinPartition";
+  addDivInside(partitionDivId, gridRow.id);
+  var partitionCol = document.getElementById(partitionDivId);
+  partitionCol.classList.add("col");
+  partitionCol.classList.add("s3");
```
Comment (kaczmarj) [inline main/js/afterSubmit.js:203]: minor thing but we can add multiple classes with one call to `.add()`

```suggestion
  partitionCol.classList.add("col", "s3");
```

File: main/js/afterSubmit.js [hunk 2]
```diff
@@ -187,16 +187,28 @@ const buildHeatmap = function(expData, clinData) {
...
 
-  // Create partition selector
-  const partitionDivId = "violinPartition";
-  addDivInside(partitionDivId, "violinLoaderDiv");
-  // Create partition selector
-  createViolinPartitionBox("violinPlots", geneQuery);
+  //Setup Materialize Grid
+  addDivInside("violinGridRow", violinLoaderDiv.id);
+  var gridRow = document.getElementById("violinGridRow");
+  gridRow.classList.add("row");
```
Comment (kaczmarj) [inline main/js/afterSubmit.js:196]: does this `row` class come from materialize-grid?

File: main/js/afterSubmit.js [hunk 3]
```diff
@@ -187,16 +187,28 @@ const buildHeatmap = function(expData, clinData) {
...
+  const partitionDivId = "violinPartition";
+  addDivInside(partitionDivId, gridRow.id);
+  var partitionCol = document.getElementById(partitionDivId);
+  partitionCol.classList.add("col");
+  partitionCol.classList.add("s3");
+  //Generate the partition selector
+  createViolinPartitionBox(partitionDivId, geneQuery);
+
+  // Create div for violin plots and add it inside Materialize Grid
+  addDivInside("violinPlots", gridRow.id);
```
Comment (kaczmarj) [inline main/js/afterSubmit.js:208]: what's the difference between the `violinPlots` and the `violinPartition` divs?

File: main/js/afterSubmit.js [hunk 4]
```diff
@@ -187,16 +187,28 @@ const buildHeatmap = function(expData, clinData) {
...
+  var violinPlotsCol = document.getElementById("violinPlots");
+  violinPlotsCol.classList.add("col")
+  violinPlotsCol.classList.add("s8");
```
Comment (kaczmarj) [inline main/js/afterSubmit.js:211]: similar comment to above

```suggestion
  violinPlotsCol.classList.add("col", "s8");
```

**Review by kaczmarj — COMMENTED** [review]
overall looks good. just had some feedback and questions.

File: main/js/afterSubmit.js (hunk 3 above)
Comment (19aanand) [reply main/js/afterSubmit.js:207]: The violinPartition div is where the partition selector exists, and the violinPlots div is where all the violin plots exist (minus the partition selector).

**Comment by 19aanand** [comment]
@amaraim22 This issues with this PR have been resolved!

**Review by prvshenoy01 — APPROVED** [review]
Looks good to me!

One observation, some of the longer clinical variable labels for the heatmap sample tracks are getting cut off because of the partition selector. This would be low priority but can the sample track labels be brought to the right side?
image

**Comment by 19aanand** [comment]
@prvshenoy01 Moving it to the right side made the amount of text displayed slightly better, but the cutoff was still noticeable for longer variable names. I changed it to the right side for now so users can see the start of the variable name.

**Pull request merged**